ts/att/time.ts

```diff
@@ -1,17 +1,70 @@
+import {
+    TimeSpan,
+} from "./timeSpan";
 export class Time {
 
     public static CreateTime24Hour(hour: number, minute: number): Time {
         return new Time(hour, minute);
     }
 
     public static CreateTime12Hour(hour: number, minute: number, isPm: boolean): Time {
-        if (hour < 1 || hour > 12) {throw new RangeError("Expected value is between 1 to 12, but actual is {hour}"); }
-        if (isPm && hour !== 12) {hour += 12; }
-        if (!isPm && hour === 12) {hour = 0; }
+        if (hour < 1 || hour > 12) {
+            throw new RangeError("Expected value is between 1 to 12, but actual is {hour}");
+        }
+        if (isPm && hour !== 12) {
+            hour += 12;
+        }
+        if (!isPm && hour === 12) {
+            hour = 0;
+        }
         const t = Time.CreateTime24Hour(hour, minute);
         return t;
     }
 
+    public static CreateTimeFrom12Hour(numbers: string, amOrpm: string): Time {
+        const tokens = numbers.split(":");
+        let hour = parseInt(tokens[0], 10);
+        const minute = parseInt(tokens[1], 10);
+
+        if (amOrpm.toLowerCase() === "pm" && hour !== 12) {
+            hour += 12;
+        }
+        if (amOrpm.toLowerCase() === "am" && hour === 12) {
+            hour = 0;
+        }
+
+        return Time.CreateTime24Hour(hour, minute);
+    }
+
+    public static CreateTimeFrom12Hour_2(hour: number, minute: number, isPm: boolean): Time {
```

Comment:
Can you rename this method to `CreateTime12Hour` ?

Reply:
I already have a method with that name and Typescript don't support method overloading

Reply:
@DheenodaraRao I know, but the other one is `CreateTimeFrom12Hour` right? The new name I requested is `CreateTime12Hour` which don't have the word  `From`.

Reply:
There is already a method with that name.

Reply:
@DheenodaraRao Sorry I didn't see that